flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -292,15 +304,39 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
         }
 
         // The MCDU is a client to the MCDU Server and tries to connect in regular intervals.
-        // every 2500ms
-        if (this.mcduServerConnectUpdateThrottler.canUpdate(_deltaTime) !== -1
-            && (!this.socket || this.socket.readyState !== 1)) {
-
-            this.connectWebsocket(NXDataStore.get("CONFIG_EXTERNAL_MCDU_PORT", "8380"));
+        // Due to an issue with the sim's Coherent engine we need to avoid trying
+        // to connect the websocket continuously. The below solution based on an EFB setting
+        // and a maximal number of attempts should mitigate the issue until
+        // Asobo has fixed the core issue.
+        if (this.mcduServerConnectUpdateThrottler.canUpdate(_deltaTime) !== -1) {
+            // Try to connect websocket if enabled in EFB and no connection established
+            this.mcduServerConnect = NXDataStore.get("CONFIG_EXTERNAL_MCDU_SERVER_ENABLED", 'AUTO ON');
+            if (this.mcduServerConnect === 'AUTO ON' && (!this.socket || this.socket.readyState !== 1)) {
+                // We try to connect for a fixed amount of attempts, then we deactivate the connection setting
+                if (this.socketConnectionAttempts++ >= this.maxConnectionAttempts) {
+                    console.log("Maximum number of connection attempts to MCDU Server exceeded. No more attempts.");
+                    NXDataStore.set("CONFIG_EXTERNAL_MCDU_SERVER_ENABLED", 'AUTO OFF');
+                    this.socketConnectionAttempts = 0;
+                } else {
+                    console.log(`Attempting MCDU Server connection ${this.socketConnectionAttempts} of ${this.maxConnectionAttempts} attempts.`);
+                    this.connectWebsocket(NXDataStore.get("CONFIG_EXTERNAL_MCDU_PORT", "8380"));
+                }
+            } else if (this.mcduServerConnect !== 'AUTO ON') {
+                if (this.socketConnectionAttempts > 0) {
+                    console.log("MCDU server connection attempts deactivated. No more attempts.");
+                    this.socketConnectionAttempts = 0;
+                }
+                if (this.socket && this.socket.readyState === 1) {
```

Comment:
Shouldn't this close the socket independent from its state? Just to be absolutely sure.

Reply:
Yes it should. Missed it. Thanks - will fix.